chain-indexer/src/domain/ledger_state.rs

```diff
@@ -49,20 +49,32 @@ impl LedgerState {
         for (variant, transaction) in transactions {
             match variant {
                 TransactionVariant::Regular => {
-                    self.apply_regular_transaction(
-                        transaction,
-                        block_parent_hash,
-                        block_timestamp,
-                    )?;
+                    self.apply_regular_transaction(transaction, block_parent_hash, block_timestamp)
+                        .map_err(|source| Error::ApplyTransaction {
+                            variant: "regular".to_string(),
+                            hash: "unknown (stored transaction)".to_string(),
+                            source,
+                        })?;
                 }
 
                 TransactionVariant::System => {
-                    self.apply_system_transaction(transaction, block_timestamp)?;
+                    self.apply_system_transaction(transaction, block_timestamp)
+                        .map_err(|source| Error::ApplyTransaction {
+                            variant: "system".to_string(),
+                            hash: "unknown (stored transaction)".to_string(),
+                            source,
+                        })?;
                 }
             }
         }
 
-        let ledger_parameters = self.post_apply_transactions(block_timestamp)?;
+        let ledger_parameters =
+            self.post_apply_transactions(block_timestamp)
+                .map_err(|source| Error::ApplyTransaction {
```

Comment:
This is nonsense. You are wrapping an expressive Error::BlockLimitExceeded in Error::ApplyTransaction. Revert this change.